snapcraft/internal/elf.py

```diff
@@ -91,6 +104,65 @@ def __init__(self, *, path: str, magic: str) -> None:
         self.path = path
         self.is_executable = 'interpreter' in magic
         self.dependencies = set()  # type: Set[Library]
+        self.symbols = self._get_symbols(path)  # type: List[Symbol]
+
+    def _get_symbols(self, path) -> List[Symbol]:
+        symbols = list()  # type: List[Symbol]
+        symbols_section = subprocess.check_output([
+            'readelf', '--wide', '--dyn-syms', path]).decode()
+        # Regex inspired by the regex in lintian to match entries similar to
+        # number '1' from the following sample output
+        #
+        # Symbol table '.dynsym' contains 2281 entries:
+        #   Num:    Value          Size Type    Bind   Vis      Ndx Name
+        #     0: 0000000000000000     0 NOTYPE  LOCAL  DEFAULT  UND
+        #     1: 0000000000000000     0 FUNC    GLOBAL DEFAULT  UND endgrent@GLIBC_2.2.5 (2)  # noqa
+        symbol_match = (r'\s*\d+:\s*[0-9a-f]+\s*\d+\s*\S+\s*\S+\s*\S+\s*'
+                        r'(?P<section>\S+)\s*(?P<symbol>\S+).*\Z')
+        regex = re.compile(symbol_match)
+        for line in symbols_section.split('\n'):
+            m = regex.search(line)
+            if m:
+                section = m.group('section')
+                try:
+                    name, version = m.group('symbol').split('@')
+                except ValueError:
+                    name = m.group('symbol')
```

Comment:
You're calling `m.group('symbol')` twice, potentially. It would be more efficient to just call it once before trying to split.